protocol/app/datadog.go

```diff
@@ -111,3 +115,42 @@ func initDatadogProfiler(logger log.Logger, ddAgentHost string, ddAgentPort uint
 		panic(err)
 	}
 }
+
+type DatadogErrorTrackingObject struct {
+	Stack   []map[string]string
+	Message string
+	Kind    string
+}
+
+func (obj DatadogErrorTrackingObject) MarshalZerologObject(e *zerolog.Event) {
+	e.Interface("stack", obj.Stack).
+		Str("message", obj.Message).
+		Str("kind", obj.Kind)
+}
+
+var (
+	zerologFormatterOnce sync.Once
+)
+
+// SetZerologDatadogErrorTrackingFormat sets custom error formatting for log tag
+// values that are errors for the zerolog library. Converts them to a format that
+// is compatible with datadog error tracking.
+func SetZerologDatadogErrorTrackingFormat() {
```

Comment:
Please add a test that captures the logs from zerolog for an error to ensure that the stack traces doesn't lose information (or add useless details) since the stack depth can change and it is dependent on the libraries implementation.

Reply:
As discussed, linear ticket [here](https://linear.app/dydx/issue/CLOB-1049/add-test-to-datadog-error-tracking-formatter-stack-trace-truncater) and added TODO.